### PR TITLE
Revamp the operator CLI for opt-out

### PR DESCRIPTION
**Pull request by jianoaix**
## Why are these changes needed?
We settle on the operator CLI semantics like this:
- opt-in quorum-list: add quorum-list to the operator (quorum-list must be not yet opted in)
- opt-out quorum-list: remove quorum-list from the operator (quorum-list must be already opted in)

## Checks

- [x] I've made sure the lint is passing in this PR.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: core/eth/tx.go [hunk 1]
```diff
@@ -280,26 +282,35 @@ func (t *Transactor) RegisterOperatorWithChurn(
...
 }
 
-// DeregisterOperator deregisters an operator with the given public key from the all the quorums that it is
+// DeregisterOperator deregisters an operator with the given public key from the specified the quorums that it is
 // registered with at the supplied block number. To fully deregister an operator, this function should be called
 // with the current block number.
-func (t *Transactor) DeregisterOperator(ctx context.Context, pubkeyG1 *core.G1Point, blockNumber uint32) error {
+// If the operator isn't registered with any of the specified quorums, this function will return error, and
+// no quorum will be deregistered.
+func (t *Transactor) DeregisterOperator(ctx context.Context, pubkeyG1 *core.G1Point, blockNumber uint32, quorumIds []core.QuorumID) error {
```
Comment (ian-shim) [inline core/eth/tx.go:290]: validate the length of `quorumIDs`?

File: node/plugin/config.go [hunk 2]
```diff
@@ -66,7 +66,7 @@ var (
 	}
 	QuorumIDListFlag = cli.StringFlag{
 		Name:     "quorum-id-list",
-		Usage:    "Comma separated list of quorum IDs that the node will participate in. There should be at least one quorum ID. This list can contain quorums node is already registered with. If the node opts in to quorums already registered with, it will be a no-op.",
+		Usage:    "Comma separated list of quorum IDs that the node will opt-in or opt-out, depending on the OperationFlag. If OperationFlag is opt-in, all quorums should not have been registered already; if it's opt-out, all quorums should have been registered already",
```
Comment (ian-shim) [inline node/plugin/config.go:69]: `all quorums should not have been registered already` sounds like they can't provide quorums that they're already registered with, which isn't true. They _can_ include quorum IDs they've already opted in

Comment (jianoaix) [reply node/plugin/config.go:69]: I'm inclined to align with smart contract API which fails the call if "register quorums that are already registered" or "deregister quorums that are not currently registered" (@0x0aa0 please correct if this isn't the case).

Comment (jianoaix) [reply node/plugin/config.go:69]: We should actually also make opt-in/opt-out consistent w.r.t. this semantics

File: core/eth/tx.go [hunk 3]
```diff
@@ -280,26 +282,38 @@ func (t *Transactor) RegisterOperatorWithChurn(
...
+	if len(quorumIds) == 0 {
```
Comment (ian-shim) [inline core/eth/tx.go:291]: Can we dedup them as well? Just wanted to make sure we don't loop over a massive slice on :305

Comment (jianoaix) [reply core/eth/tx.go:291]: That doesn't seem to help much, no matter it's dup or no dup, it's provided by users who are going to run this cmd

Comment (ian-shim) [reply core/eth/tx.go:291]: Sure, how about just check `len(quorumIDs) < 256`? It would prevent them from passing a huge number of elems

Comment (ian-shim) [reply core/eth/tx.go:291]: still to be done

**Review by ian-shim — APPROVED** [review]
lgtm. one remaining comment on slice length validation

File: core/eth/tx.go (hunk 3 above)
Comment (jianoaix) [reply core/eth/tx.go:291]: https://github.com/Layr-Labs/eigenda/pull/381/commits/1629ea6e15a843a9bd3eb58e38bb7652a6da81e6

**Pull request merged**